x/ecocredit/migrations/v3/core.go

```diff
@@ -0,0 +1,404 @@
+		classNameIDMap[classInfo.ClassId] = classID
+
+		// migrate class issuers to ORM v1
+		for _, issuer := range classInfo.Issuers {
+			addr, err := sdk.AccAddressFromBech32(issuer)
+			if err != nil {
+				return err
+			}
+
+			if err := ss.ClassIssuerTable().Insert(ctx, &api.ClassIssuer{
+				ClassId: classID,
+				Issuer:  addr,
+			}); err != nil {
+				return err
+			}
+		}
+
+		// delete class info from old store
+		if err := classInfoTable.Delete(sdkCtx, &ClassInfo{ClassId: classInfo.ClassId}); err != nil {
+			return err
+		}
+	}
+
+	// migrate credit type sequence to ORM v1
+	cItr, err := creditTypeSeqTable.PrefixScan(sdkCtx, nil, nil)
+	if err != nil {
+		return err
+	}
+	defer cItr.Close()
+
+	for {
+		var ctype CreditTypeSeq
+		if _, err := cItr.LoadNext(&ctype); err != nil {
+			if orm.ErrIteratorDone.Is(err) {
+				break
+			}
+
+			return err
+		}
+		if err := ss.ClassSequenceTable().Save(ctx, &api.ClassSequence{
+			CreditType:  ctype.Abbreviation,
+			NextClassId: ctype.SeqNumber,
+		}); err != nil {
+			return err
+		}
+
+		// delete credit type sequence from old store
+		if err := creditTypeSeqTable.Delete(sdkCtx, &ctype); err != nil {
+			return err
+		}
+	}
+
+	// migrate credit batches to ORM v1 and create projects for existing credit classes
+	batchIDsMap := make(map[string]batchMapT) // map of a batch denom to batch-id and amount cancelled
+	batchSeqMap := make(map[uint64]uint64)    // map of a project-id to batch sequence
+	projectSeqMap := make(map[uint64]uint64)  // map of a credit classID to project sequence
+	batchItr, err := batchInfoTable.PrefixScan(sdkCtx, nil, nil)
+	if err != nil {
+		return err
+	}
+	defer batchItr.Close()
+
+	for {
+		var batchInfo BatchInfo
+		if _, err := batchItr.LoadNext(&batchInfo); err != nil {
+			if orm.ErrIteratorDone.Is(err) {
+				break
+			}
+			return err
+		}
+
+		admin, err := sdk.AccAddressFromBech32(batchInfo.Issuer)
+		if err != nil {
+			return err
+		}
+		pItr, err := ss.ProjectInfoTable().List(ctx, api.ProjectInfoAdminIndexKey{}.WithAdmin(admin.Bytes()))
+		if err != nil {
+			return err
+		}
+
+		projectExists := false
+		var projectID uint64
+		for pItr.Next() {
+			pInfo, err := pItr.Value()
+			if err != nil {
+				return err
+			}
+
+			if pInfo.ClassId == classNameIDMap[batchInfo.ClassId] {
+				projectExists = true
+				projectID = pInfo.Id
+				break
+			}
+
+		}
+		pItr.Close()
+
+		if !projectExists {
+			classID := classNameIDMap[batchInfo.ClassId]
+			var projectSeq uint64 = 1
+			if val, ok := projectSeqMap[classID]; ok {
+				projectSeqMap[classID] = val + 1
+				projectSeq = val
+			} else {
+				projectSeqMap[classID] = 2
+			}
+
+			name := FormatProjectID(batchInfo.ClassId, projectSeq)
+			id, err := ss.ProjectInfoTable().InsertReturningID(ctx,
+				&api.ProjectInfo{
+					Name:            name,
+					Admin:           admin,
+					ClassId:         classID,
+					ProjectLocation: batchInfo.ProjectLocation,
+					Metadata:        "", // TODO: add metadata
```

Comment:
I think we can remove this todo or replace it with a comment about metadata being handled in the upgrade handler on a per chain basis.